#' @name plot_hist
#'
#' @title This function prints plots for all plots given by plot_hist()
#'
#' @param SangeR object from plot_hist function.
#'
#' @return no return, prints PNGs with histograms in working directory
#'
#' @export
#'


print_plots <-  function(SangeR){

  ####print mosaic

  mosaic <- gridExtra::grid.arrange(grobs = SangeR$PNG_list[!duplicated(substr(SangeR$tags,1,4))], ncol = 1)

  ggplot2::ggsave(filename = paste0(SangeR$Bnummer, "_", SangeR$genename, "_mosaic.png"), plot = mosaic)

  ###print mutated

  if(any(!SangeR$tags %in% SangeR$tags_POI)){
    mutated<- gridExtra::grid.arrange(grobs = SangeR$PNG_list[!SangeR$tags %in% SangeR$tags_POI], ncol = 1)

    ggplot2::ggsave(filename = paste0(SangeR$Bnummer, "_", SangeR$genename, "_mutated.png"), plot = mutated)
  }

  ###file for gene, block, mutation

  maf <- t(data.frame(x = rep(234,x = "NA")))
  colnames(maf) <- c("Hugo_Symbol",
                     "Entrez_Gene_Id",
                     "Center",
                     "NCBI_Build",
                     "Chromosome",
                     "Start_Position",
                     "End_Position",
                     "Strand",
                     "Variant_Classification",
                     "Variant_Type",
                     "Reference_Allele",
                     "Tumor_Seq_Allele1",
                     "Tumor_Seq_Allele2",
                     "dbSNP_RS",
                     "dbSNP_Val_Status",
                     "Tumor_Sample_Barcode",
                     "Matched_Norm_Sample_Barcode",
                     "Match_Norm_Seq_Allele1",
                     "Match_Norm_Seq_Allele2",
                     "Tumor_Validation_Allele1",
                     "Tumor_Validation_Allele2",
                     "Match_Norm_Validation_Allele1",
                     "Match_Norm_Validation_Allele2",
                     "Verification_Status",
                     "Validation_Status",
                     "Mutation_Status",
                     "Sequencing_Phase",
                     "Sequence_Source",
                     "Validation_Method",
                     "Score",
                     "BAM_File",
                     "Sequencer",
                     "Tumor_Sample_UUID",
                     "Matched_Norm_Sample_UUID",
                     "Genome_Change",
                     "Annotation_Transcript",
                     "Transcript_Strand",
                     "Transcript_Exon",
                     "Transcript_Position",
                     "c",
                     "NA_Change",
                     "Codon_Change",
                     "Protein_Change",
                     "Other_Transcripts",
                     "Refseq_m",
                     "NA_Id",
                     "Refseq_prot_Id",
                     "SwissProt_acc_Id",
                     "SwissProt_entry_Id",
                     "Description",
                     "UniProt_AApos",
                     "UniProt_Region",
                     "UniProt_Site",
                     "UniProt",
                     "Natural_Variations",
                     "UniProt_Experimental_Info",
                     "GO_Biological_Process",
                     "GO_Cellular_Component",
                     "GO_Molecular_Function",
                     "COSMIC_overlapping_mutations",
                     "COSMIC_fusion_genes",
                     "COSMIC_tissue_types_affected",
                     "COSMIC_total_alterations_in_gene",
                     "Tumorscape_Amplification_Peaks",
                     "Tumorscape_Deletion_Peaks",
                     "TCGAscape_Amplification_Peaks",
                     "TCGAscape_Deletion_Peaks",
                     "DrugBank",
                     "ref_context",
                     "gc_content",
                     "CCLE_ONCOMAP_overlapping_mutations",
                     "CCLE_ONCOMAP_total_mutations_in_gene",
                     "CGC_Mutation_Type",
                     "CGC_Translocation_Partner",
                     "CGC_Tumor_Types_Somatic",
                     "CGC_Tumor_Types_Germline",
                     "CGC_Other_Diseases",
                     "NARepairGenes_Activity_linked_to_OMIM",
                     "FamilialCancerDatabase_Syndromes",
                     "MUTSIG_Published_Results",
                     "OREGANNO_ID",
                     "OREGANNO_Values",
                     "tumor_f",
                     "t_alt_count",
                     "t_ref_count",
                     "n_alt_count",
                     "n_ref_count",
                     "Gencode_19_secondaryVariantClassification",
                     "Achilles_Top_Genes",
                     "CGC",
                     "Name",
                     "CGC_GeneID",
                     "CGC_Chr",
                     "CGC_Chr_Band",
                     "CGC_Cancer_Somatic_Mut",
                     "CGC_Cancer_Germline_Mut",
                     "CGC_Cancer_Syndrome",
                     "CGC_Tissue_Type",
                     "CGC_Cancer_Molecular_Genetics",
                     "CGC_Other_Germline_Mut",
                     "ClinVar_HGMD_ID",
                     "ClinVar_SYM",
                     "ClinVar_TYPE",
                     "ClinVar_ASSEMBLY",
                     "ClinVar_rs",
                     "CosmicFusion_fusion_id",
                     "NARepairGenes_Chromosome_location_linked_to_NCBI_MapView",
                     "NARepairGenes_Accession_number_linked_to_NCBI_Entrez",
                     "Familial_Cancer_Genes_Synonym",
                     "Familial_Cancer_Genes_Reference",
                     "Gencode_XHGNC_hgnc_id",
                     "HGNC_HGNC_ID",
                     "HGNC_Status",
                     "HGNC_Locus_Type",
                     "HGNC_Locus_Group",
                     "HGNC_Previous_Symbols",
                     "HGNC_Previous",
                     "HGNC_Synonyms",
                     "HGNC",
                     "Name_Synonyms",
                     "HGNC_Chromosome",
                     "HGNC_Date_Modified",
                     "HGNC_Date_Symbol_Changed",
                     "HGNC_Date",
                     "Name_Changed",
                     "HGNC_Accession_Numbers",
                     "HGNC_Enzyme_IDs",
                     "HGNC_Ensembl_Gene_ID",
                     "HGNC_Pubmed_IDs",
                     "HGNC_RefSeq_IDs",
                     "HGNC_Gene_Family_ID",
                     "HGNC_Gene_Family",
                     "HGNC_CCDS_IDs",
                     "HGNC_Vega_ID",
                     "HGNC_OMIM_ID(supplied_by_OMIM)",
                     "HGNC_RefSeq(supplied_by_NCBI)",
                     "HGNC_UniProt_ID(supplied_by_UniProt)",
                     "HGNC_Ensembl_ID(supplied_by_Ensembl)",
                     "HGNC_UCSC_ID(supplied_by_UCSC)",
                     "Oreganno_Build",
                     "Simple_Uniprot_alt_uniprot_accessions",
                     "chr1_a_bed",
                     "chr1_a_bed_score",
                     "chr1_a_bed_strand",
                     "chr1_a_bed_thickStart",
                     "chr1_a_bed_thickEnd",
                     "chr1_a_bed_itemRgb",
                     "chr1_a_bed_blockCount",
                     "chr1_a_bed_blockSizes",
                     "chr1_a_bed_blockStarts",
                     "chr1_b_bed",
                     "chr1_b_bed_score",
                     "chr1_b_bed_strand",
                     "chr1_b_bed_thickStart",
                     "chr1_b_bed_thickEnd",
                     "chr1_b_bed_itemRgb",
                     "chr1_b_bed_blockCount",
                     "chr1_b_bed_blockSizes",
                     "chr1_b_bed_blockStarts",
                     "dbSNP_ASP",
                     "dbSNP_ASS",
                     "dbSNP_CAF",
                     "dbSNP_CDA",
                     "dbSNP_CFL",
                     "dbSNP_COMMON",
                     "dbSNP_DSS",
                     "dbSNP_G5",
                     "dbSNP_G5A",
                     "dbSNP_GENEINFO",
                     "dbSNP_GNO",
                     "dbSNP_HD",
                     "dbSNP_INT",
                     "dbSNP_KGPhase1",
                     "dbSNP_KGPhase3",
                     "dbSNP_LSD",
                     "dbSNP_MTP",
                     "dbSNP_MUT",
                     "dbSNP_NOC",
                     "dbSNP_NOV",
                     "dbSNP_NSF",
                     "dbSNP_NSM",
                     "dbSNP_NSN",
                     "dbSNP_OM",
                     "dbSNP_OTH",
                     "dbSNP_PM",
                     "dbSNP_PMC",
                     "dbSNP_R3",
                     "dbSNP_R5",
                     "dbSNP_REF",
                     "dbSNP_RV",
                     "dbSNP_S3D",
                     "dbSNP_SAO",
                     "dbSNP_SLO",
                     "dbSNP_SSR",
                     "dbSNP_SYN",
                     "dbSNP_TOPMED",
                     "dbSNP_TPA",
                     "dbSNP_U3",
                     "dbSNP_U5",
                     "dbSNP_VC",
                     "dbSNP_VP",
                     "dbSNP_WGT",
                     "dbSNP_WTD",
                     "dbSNP_dbSNPBuildID",
                     "dbSNP_ID",
                     "dbSNP_FILTER",
                     "HGNC_Entrez_Gene_ID(supplied_by_NCBI)",
                     "dbSNP_RSPOS",
                     "dbSNP_VLD",
                     "AS_FilterStatus",
                     "AS_SB_TABLE",
                     "AS_UNIQ_ALT_READ_COUNT",
                     "CONTQ",
                     "DP",
                     "ECNT",
                     "GERMQ",
                     "MBQ",
                     "MFRL",
                     "MMQ",
                     "MPOS",
                     "NALOD",
                     "NCount",
                     "NLOD",
                     "OCM",
                     "PON",
                     "POPAF",
                     "ROQ",
                     "RPA",
                     "RU",
                     "SEQQ",
                     "STR",
                     "STRANDQ",
                     "STRQ",
                     "TLOD")

  utils::write.csv2(quote = FALSE, row.names =  FALSE, x = c(SangeR$Bnummer, SangeR$genename, SangeR$tags[!SangeR$tags %in% SangeR$tags_POI]), file = paste0(SangeR$Bnummer, "_", SangeR$genename, ".txt"))

}
